src/components/MainView/main-view.jsx

```diff
@@ -45,22 +46,34 @@ export class MainView extends Component {
    */
   async componentDidMount() {
     this.props.setLoadingStatus(true);
+    this.props.setHook(localStorage.getItem('PERSISTED_hook') || 'patient-view');
     await smartLaunchPromise().catch(async () => {
-      await retrieveFhirMetadata().catch(() => new Promise((resolve) => {
+      const persistedFhirServer = localStorage.getItem('PERSISTED_fhirServer');
+      await retrieveFhirMetadata(persistedFhirServer).catch(() => new Promise((resolve) => {
         this.setState({
           fhirServerPrompt: true,
           fhirServerPromptHold: resolve,
         });
       }));
     });
     if (this.state.fhirServerPrompt) { this.setState({ fhirServerPrompt: false }); }
-    await retrievePatient().catch(() => new Promise((resolve) => {
+    const persistedPatientId = localStorage.getItem('PERSISTED_patientId');
+    await retrievePatient(persistedPatientId).catch(() => new Promise((resolve) => {
       this.setState({
         patientPrompt: true,
         patientPromptHold: resolve,
       });
     }));
     if (this.state.patientPrompt) { this.setState({ patientPrompt: false }); }
+    const persistedServices = localStorage.getItem('PERSISTED_cdsServices');
+    if (persistedServices) {
```

Comment:
Nit: You could check ```persistedServices.length``` here so that we wouldnt need to parse if the length is 0.

Reply:
We parse it because the localStorage will return JSON stringified so we can check if the property exists yet in localStorage, then pass if not. Then check if the parsed JSON string returns an empty array or not.